source/dotnet/wholesale-api/Infrastructure/EventPublishers/IntegrationEventTypeMapper.cs

```diff
@@ -18,9 +18,13 @@ public class IntegrationEventTypeMapper : IIntegrationEventTypeMapper
 {
     private readonly Dictionary<Type, string> _dic;
 
-    public IntegrationEventTypeMapper()
+    public IntegrationEventTypeMapper(Dictionary<Type, string> dic)
     {
         _dic = new Dictionary<Type, string>();
+        foreach (var key in dic)
+        {
+            Add(key.Value, key.Key);
+        }
     }
```

Comment:
How about simplifying things with this + removing the `Add` method?
```suggestion
    public IntegrationEventTypeMapper(Dictionary<Type, string> dic)
    {
        _dic = new Dictionary<Type, string>(dic);
    }
```

Reply:
If I remove the Add the validation is fired.